test/test_profiler.py

```diff
@@ -1166,16 +1166,25 @@ def linked_correlation_id(self) -> int:
                 return self._linked_correlation_id
 
             def device_type(self) -> DeviceType:
-                return self._device_type
+                return DeviceType.CUDA if self._device_type == 1 else DeviceType.CPU
 
-        @dataclass(frozen=True)
         class MockProfilerEvent():
-            _name: str
-            id: int
-            start_time_ns: int
-            duration_time_ns: int
-            children = []
-            parent = None
+
+            def __init__(self,
```

Comment:
You don't have to give up the dataclass. The idiomatic way to do non-trivial init on a frozen class is to use `__post_init__` and `object.__set_attr__`.

Example:
https://github.com/pytorch/pytorch/blob/c5f9e043e96f5c60e97292855572576335002c8a/benchmarks/instruction_counts/core/api.py#L62-L66